### PR TITLE
Fix #112: Introduce question progress controller interface

### LINKED ISSUE #112: Introduce interface for QuestionAssessmentProgressController
This is tracking introducing a stubbed interface for #111 without the real implementation being complete.

### PR DESCRIPTION
**Pull request by BenHenning**
This introduces only the interface for the question progress controller, based on the exploration progress controller which operates similarly.

### CODE REVIEW

**Comment by BenHenning** [comment]
This is blocked on #217 and can't yet be reviewed until that PR is updated to develop and gets merged in.

**Comment by BenHenning** [comment]
@vinitamurthi & @jamesxu0 PTAL. I'll wait to merge this into develop until after #227 is merged in, but this is now a complete solution of #112.

**Comment by BenHenning** [comment]
This is now up-to-date with develop and the recent changes from #227. PTAL @jamesxu0 & @vinitamurthi.

**Review by jamesxu0 — APPROVED** [review]

File: domain/src/main/java/org/oppia/domain/question/QuestionTrainingController.kt [hunk 1]
```diff
@@ -35,13 +36,16 @@ class QuestionTrainingController @Inject constructor(
...
    */
-  fun startQuestionTrainingSession(skillIdsList: List<String>): LiveData<AsyncResult<List<Question>>> {
+  fun startQuestionTrainingSession(skillIdsList: List<String>): LiveData<AsyncResult<Any>> {
     return try {
       val retrieveQuestionsDataProvider = retrieveQuestionsForSkillIds(skillIdsList)
       questionAssessmentProgressController.beginQuestionTrainingSession(
         retrieveQuestionsDataProvider
       )
-      dataProviders.convertToLiveData(retrieveQuestionsDataProvider)
+      val erasedDataProvider: DataProvider<Any> = dataProviders.transform(
```
Comment (jamesxu0) [inline domain/src/main/java/org/oppia/domain/question/QuestionTrainingController.kt:46]: What's the point of using dataProviders.transform if it doesn't actually transform the data provider?

File: domain/src/main/java/org/oppia/domain/question/QuestionTrainingConstantsProvider.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package org.oppia.domain.question
```
Comment (vinitamurthi) [inline domain/src/main/java/org/oppia/domain/question/QuestionTrainingConstantsProvider.kt]: This file needs to be removed

File: domain/src/test/java/org/oppia/domain/question/QuestionTrainingControllerTest.kt [hunk 3]
```diff
@@ -156,12 +160,25 @@ class QuestionTrainingControllerTest {
...
+    @Mock
+    lateinit var questionTrainingConstantsProvider: QuestionTrainingConstantsProvider
+
     @Provides
     @Singleton
     fun provideContext(application: Application): Context {
       return application
     }
 
+    @Provides
```
Comment (vinitamurthi) [inline domain/src/test/java/org/oppia/domain/question/QuestionTrainingControllerTest.kt]: This is old, you can pull the changes from develop once again to update this!

File: domain/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 4]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (vinitamurthi) [inline domain/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: Remove this file

**Review by vinitamurthi — CHANGES_REQUESTED** [review]

File: domain/src/main/java/org/oppia/domain/question/QuestionTrainingController.kt (hunk 1 above)
Comment (BenHenning) [reply domain/src/main/java/org/oppia/domain/question/QuestionTrainingController.kt:46]: Good question--this also threw me off looking back on this. I believe it's due to needing to change the type to 'Any' to erase it for the return value. Clarified via a comment.

File: domain/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 4 above)
Comment (BenHenning) [reply domain/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: Weird, I don't see this file locally. Let's see if it disappears upon my next push.

File: domain/src/test/java/org/oppia/domain/question/QuestionTrainingControllerTest.kt (hunk 3 above)
Comment (BenHenning) [reply domain/src/test/java/org/oppia/domain/question/QuestionTrainingControllerTest.kt]: Done!

File: domain/src/main/java/org/oppia/domain/question/QuestionTrainingConstantsProvider.kt (hunk 2 above)
Comment (BenHenning) [reply domain/src/main/java/org/oppia/domain/question/QuestionTrainingConstantsProvider.kt]: Looks removed after merging in develop.

**Review by vinitamurthi — APPROVED** [review]

**Comment by BenHenning** [comment]
Thanks!

**Pull request merged**